python/lsst/skymap/baseSkyMap.py

```diff
@@ -282,30 +282,32 @@ def register(self, name, registry):
             nx, ny = tractInfo.getNumPatches()
             nxMax = max(nxMax, nx)
             nyMax = max(nyMax, ny)
-        registry.addDimensionEntry(
-            "SkyMap",
-            {"skymap": name,
-             "hash": self.getSha1(),
-             "tract_max": len(self),
-             "patch_nx_max": nxMax,
-             "patch_ny_max": nyMax}
-        )
-        for tractInfo in self:
-            region = tractInfo.getOuterSkyPolygon()
-            centroid = SpherePoint(region.getCentroid())
+        with registry.transaction():
             registry.addDimensionEntry(
-                "Tract",
-                {"skymap": name, "tract": tractInfo.getId(),
-                 "region": region,
-                 "ra": centroid.getRa().asDegrees(),
-                 "dec": centroid.getDec().asDegrees()}
+                "SkyMap",
+                {"skymap": name,
+                 "hash": self.getSha1(),
+                 "tract_max": len(self),
+                 "patch_nx_max": nxMax,
+                 "patch_ny_max": nyMax}
             )
-            for patchInfo in tractInfo:
-                cellX, cellY = patchInfo.getIndex()
-                registry.addDimensionEntry(
-                    "Patch",
+            for tractInfo in self:
+                region = tractInfo.getOuterSkyPolygon()
+                centroid = SpherePoint(region.getCentroid())
+                entry = (
                     {"skymap": name, "tract": tractInfo.getId(),
-                     "patch": tractInfo.getSequentialPatchIndex(patchInfo),
-                     "cell_x": cellX, "cell_y": cellY,
-                     "region": patchInfo.getOuterSkyPolygon(tractInfo.getWcs())}
+                     "region": region,
+                     "ra": centroid.getRa().asDegrees(),
+                     "dec": centroid.getDec().asDegrees()}
                 )
+                registry.addDimensionEntry("Tract", entry)
```

Comment:
Tracts must all be present before patches can be added, but I found that if all the tracts were added up front there were issues with populating some of the join tables. Since there are many many fewer tracts, I kept adding them singly before each vectorized call to add the patches.